libcudacxx/include/cuda/__type_traits/is_floating_point.h

```diff
@@ -0,0 +1,55 @@
+//===----------------------------------------------------------------------===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+// SPDX-FileCopyrightText: Copyright (c) 2025 NVIDIA CORPORATION & AFFILIATES.
+//
+//===----------------------------------------------------------------------===//
+
+#ifndef __CUDA__TYPE_TRAITS_IS_FLOATING_POINT_H
+#define __CUDA__TYPE_TRAITS_IS_FLOATING_POINT_H
+
+#include <cuda/std/detail/__config>
+
+#if defined(_CCCL_IMPLICIT_SYSTEM_HEADER_GCC)
+#  pragma GCC system_header
+#elif defined(_CCCL_IMPLICIT_SYSTEM_HEADER_CLANG)
+#  pragma clang system_header
+#elif defined(_CCCL_IMPLICIT_SYSTEM_HEADER_MSVC)
+#  pragma system_header
+#endif // no system header
+
+#include <cuda/std/__type_traits/integral_constant.h>
+#include <cuda/std/__type_traits/is_extended_floating_point.h>
+#include <cuda/std/__type_traits/is_floating_point.h>
+#include <cuda/std/__type_traits/remove_cv.h>
+
+_LIBCUDACXX_BEGIN_NAMESPACE_CUDA
+
+template <class _Tp>
+struct __cccl_is_floating_point : _CUDA_VSTD::is_floating_point<_Tp>
+{};
+#ifdef _LIBCUDACXX_HAS_NVFP16
+template <>
+struct __cccl_is_floating_point<__half> : _CUDA_VSTD::true_type
+{};
+#endif // _LIBCUDACXX_HAS_NVFP16
+#ifdef _LIBCUDACXX_HAS_NVBF16
+template <>
+struct __cccl_is_floating_point<__nv_bfloat16> : _CUDA_VSTD::true_type
+{};
+#endif // _LIBCUDACXX_HAS_NVBF16
+
+template <class _Tp>
+struct _CCCL_TYPE_VISIBILITY_DEFAULT is_floating_point : __cccl_is_floating_point<_CUDA_VSTD::remove_cv_t<_Tp>>
+{};
```

Comment:
Should we rather define this as:
```suggestion
template <class _Tp>
using is_floating_point : bool_constant<_CUDA_VSTD::is_floating_point<_Tp>::value || _CUDA_VSTD::is_extended_floating_point<_Tp>::value>
{};
```
? As suggested by @davebayer.

Reply:
It does not really matter

Reply:
I will change the implementation, because it also ensures that `is_extended_floating_point` stays consistent with `cuda::is_floating_point`.